core/trino-web-ui/src/main/resources/webapp-preview/src/components/Dashboard.tsx

```diff
@@ -11,119 +11,182 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { useEffect, useState } from 'react';
 import Typography from '@mui/material/Typography';
 import {
   Box,
-  Button,
-  Checkbox,
   Grid2 as Grid,
-  Link,
-  Paper,
-  Stack,
-  Switch,
-  Table,
-  TableBody,
-  TableCell,
-  TableContainer,
-  TableHead,
-  TableRow
 } from "@mui/material";
+import { MetricCard } from './MetricCard.tsx'
+import { useSnackbar } from "./SnackbarContext.ts";
+import { statsApi } from "../api/webapp/api.ts";
+import { Texts } from "../constant.ts";
+import {
+  MAX_HISTORY,
+  addExponentiallyWeightedToHistory,
+  addToHistory,
+  formatCount,
+  formatDataSizeBytes,
+  precisionRound,
+} from '../utils/utils.ts'
+
+interface ClusterStats
+{
+  runningQueries: number[];
+  queuedQueries: number[];
+  blockedQueries: number[];
+  activeWorkers: number[];
+  runningDrivers: number[];
+  reservedMemory: number[];
+  rowInputRate: number[];
+  byteInputRate: number[];
+  perWorkerCpuTimeRate: number[];
+
+  lastRefresh: number | null;
+
+  lastInputRows: number
+  lastInputBytes: number;
+  lastCpuTime: number;
+}
 
 export const Dashboard = () => {
-  const createData = (
-    name: string,
-    calories: number,
-    fat: number,
-    carbs: number,
-    protein: number,
-  ) => {
-    return { name, calories, fat, carbs, protein };
-  }
-
-  const rows = [
-    createData('Frozen yoghurt', 159, 6.0, 24, 4.0),
-    createData('Ice cream sandwich', 237, 9.0, 37, 4.3),
-    createData('Eclair', 262, 16.0, 24, 6.0),
-  ];
+  const { showSnackbar } = useSnackbar();
+  const initialFilledHistory = Array(MAX_HISTORY).fill(0);
+  const [clusterStats, setClusterStats] = useState<ClusterStats>({
+    runningQueries: initialFilledHistory,
+    queuedQueries: initialFilledHistory,
+    blockedQueries: initialFilledHistory,
+    activeWorkers: initialFilledHistory,
+    runningDrivers: initialFilledHistory,
+    reservedMemory: initialFilledHistory,
+    rowInputRate: initialFilledHistory,
+    byteInputRate: initialFilledHistory,
+    perWorkerCpuTimeRate: initialFilledHistory,
+
+    lastRefresh: null,
+
+    lastInputRows: 0,
+    lastInputBytes: 0,
+    lastCpuTime: 0,
+  });
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    getClusterStats();
+    const intervalId = setInterval(getClusterStats, 1000);
+    return () => clearInterval(intervalId);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (error) {
+      showSnackbar(error, 'error');
+    }
+  }, [error, showSnackbar])
+
+  const getClusterStats = () => {
+    setError(null);
+    statsApi().then((apiResponse) => {
```

Comment:
I'd prefer the response to be a concrete class

Reply:
this is now `ApiResponse<Stats>` type.